exporter/f680/application_sections/approval_details/forms.py

```
@@ -0,0 +1,83 @@
+from django import forms
+from django.db.models import TextChoices
+from django.template.loader import render_to_string
+
+from crispy_forms_gds.layout.content import HTML
+
+from core.common.forms import BaseForm, TextChoice
+from core.forms.layouts import F680ConditionalCheckboxes, F680ConditionalCheckboxesQuestion
+
+
+class ApprovalTypeForm(BaseForm):
+    class Layout:
+        TITLE = "Select the types of approvals you need"
+        TITLE_AS_LABEL_FOR = "approval_choices"
+        SUBMIT_BUTTON_TEXT = "Save and continue"
+
+    class ApprovalTypeChoices(TextChoices):
+        INITIAL_DISCUSSIONS_OR_PROMOTING = (
+            "initial_discussion_or_promoting",
+            "Initial discussions or promoting products",
+        )
+        DEMONSTRATION_IN_THE_UK = (
+            "demonstration_in_uk",
+            "Demonstration in the United Kingdom to overseas customers",
+        )
+        DEMONSTRATION_OVERSEAS = "demonstration_overseas", "Demonstration overseas"
+        TRAINING = "training", "Training"
+        THROUGH_LIFE_SUPPORT = "through_life_support", "Through life support"
+        SUPPLY = "supply", "Supply"
+
+    ApprovalTypeChoices = (
+        TextChoice(ApprovalTypeChoices.INITIAL_DISCUSSIONS_OR_PROMOTING),
+        TextChoice(ApprovalTypeChoices.DEMONSTRATION_IN_THE_UK),
+        TextChoice(ApprovalTypeChoices.DEMONSTRATION_OVERSEAS),
+        TextChoice(ApprovalTypeChoices.TRAINING),
+        TextChoice(ApprovalTypeChoices.THROUGH_LIFE_SUPPORT),
+        TextChoice(ApprovalTypeChoices.SUPPLY),
+    )
+
+    approval_choices = forms.MultipleChoiceField(
+        choices=(),
+        error_messages={
+            "required": "Select an approval choice",
+        },
+        widget=forms.CheckboxSelectMultiple(),
+    )
```

This is missing a label attribute, which we need to ensure that the question itself gets populated in the JSON.
Would recommend making it `label=Layout.TITLE`, to keep things consistent.

This does make me think that the JSON builder should raise a hard exception if `label` can't be found in order to flag this to us nice and early.

---

Yeah we need to sort that one too